### ISSUE TITLE
SessionTextureThumbnailSource

### ISSUE THREAD

**Issue by 3x1t-5tyl3** [Yellow-Dog-Man/Resonite-Issues#3130]
### Is your feature request related to a problem? Please describe.

It doesn't exist and I might not have control over the world orb.

### Describe the solution you'd like

Add a component similar to WorldTextureThumbnailSource; For a session.

### Describe alternatives you've considered

Using the SessionCaptureThumbnailSource and enveloping in a sphere some far-off distance away from users to ensure it doesn't capture stuff I don't want it to.
Which is a hacky workaround.

### Additional Context

Now; I'm somewhat sure that the session captures happen by the users?
Though I don't think that should matter as it is just a straight up texture supplied to it.

As a side solution the custom session can be made with a custom facet/orb; But that's inperfect and requires making a specific item; rather than anyone in the session being able to plop one out for someone else.

### Requesters

3x1t_5tyl3

**Comment by shiftyscales** [comment]
Can you clarify what the nature of the "stuff I don't want it to [capture]" is?

Session thumbnails by their nature are intended to provide a preview of what the activity in the world is prior to a user joining it.

E.g. if the intent is to host a session that isn't suitable for the public more publicly- this would not be an acceptable use-case- particularly since the slot providing the session thumbnail could be modified / removed, causing the content you intended to hide to be broadcast to the public.

**Comment by Dusty-Sprinkles** [comment]
Personally I added a thumbnail to my world because I don't really want to have a random front facing photo of me being a preview of what I'm doing, usually when I see those they just come off as odd so I definitely see the appeal of session specific thumbnails

**Comment by shiftyscales** [comment]
> I don't really want to have a random front facing photo of me being a preview of what I'm doing

This is why the `SessionCaptureThumbnailSource` component exists, @Dusty-Sprinkles. Resonite has no way of knowing where a good spot in a given session is to capture what is going on inside of it- if your session has a place where you / other users are commonly gathered- placing this component there will ensure a better quality / more representative thumbnail is captured.

Otherwise as a default- each user captures their own to ensure the thumbnail shows the activity of at least a user inside of the session.

**Comment by Dusty-Sprinkles** [comment]
ah, I hadn't noticed that they said that that wasn't something they wanted to use mb I guess my logic doesn't quite apply then

**Comment by shiftyscales** [comment]
> Does it matter?

It does- understanding the underlying issues you are having help determine if the solution you are asking for is suitable to the problem you are trying to highlight.

As-is the existing `SessionCaptureThumbnailSource` already has overlay functionality to add custom visuals on-top of the thumbnail- is that insufficient to meet your needs for some reason?

Documentation on these components and how to use them are available on the wiki: https://wiki.resonite.com/World_and_Session_Thumbnails#Customising_Thumbnails

**Comment by badhaloninja** [comment]
> As-is the existing `SessionCaptureThumbnailSource` already has overlay functionality to add custom visuals on-top of the thumbnail

The `SessionCaptureThumbnailSource` overlay has an explicit check to only allow images with an average alpha of 0.25 or below.

This blocks the ability to just give it your own image entirely, one of the use cases mentioned before.

**Comment by badhaloninja** [comment]
> You are still providing it your own image- it just can not cover the entirety of the visible area.

Yes, however the mentioned usecase involves **completely** overwriting the image 

Which currently you can only do by putting a sphere somewhere with the desired image and the `SessionCaptureThumbnailSource`

If you need more examples of use cases of why someone would do this,
I would use this to hide spoilers for game worlds
Or to have static thumbnail for a headless world for when no-one is in it

**Comment by shiftyscales** [comment]
Curious for input from @Frooxius - I assume there was reason why there wasn't such functionality included for sessions specifically when these components were originally created. I believe the absence of that functionality was likely originally by design.

**Comment by Dusty-Sprinkles** [comment]
This has yet to receive any reply/confirmation from froox/the team, is the needs more information tag still necessary/is this issue okay to live on?
